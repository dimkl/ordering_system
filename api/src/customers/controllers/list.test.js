/**
 * @integration-test true
 */
const Customer = require("../models/customer");

describe("GET /customers/:customer?", () => {
  beforeAll(() => require('../../shared/setupModels')());
  beforeEach(() => Customer.knex().raw('truncate customers, orders cascade;'));
  afterAll(() => Customer.knex().destroy());

  it("returns all customers", async () => {
    await Customer.query().insert({
      "first_name": "Dimitris",
      "last_name": "Klouvas",
      "email": "dimitris.klouvas@gmail.com",
      "password": "1234"
    });

    const response = await request.get("/customers")
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body.length).toBe(1)
    expect(response.body[0]).toMatchSnapshot({
      id: expect.any(Number),
      created_at: expect.any(String),
      updated_at: expect.any(String),
    });
  });

  it("returns specified customer", async () => {
    const customer = await Customer.query().insert({
      "first_name": "Dimitris",
      "last_name": "Klouvas",
      "email": "dimitris.klouvas@gmail.com",
      "password": "1234"
    });

    const response = await request.get("/customers/" + customer.id)
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({
      id: expect.any(Number),
      created_at: expect.any(String),
      updated_at: expect.any(String),
    });
  });

  it("returns empty list of customers when there are no customers", async () => {
    const response = await request.get("/customers")
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toEqual([]);
  });

  it("throws 404 when specified customer does not exist", async () => {
    const response = await request.get("/customers/" + Math.floor((Math.random() * 100)))
      .set("Accept", "application/json");

    expect(response.status).toBe(404);
  });
});