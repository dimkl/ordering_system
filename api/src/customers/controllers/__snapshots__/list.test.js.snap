// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /customers/:customer? returns all customers 1`] = `
Object {
  "email": "dimitris.klouvas@gmail.com",
  "first_name": "Dimitris",
  "id": Any<Number>,
  "last_name": "Klouvas",
}
`;

exports[`GET /customers/:customer? returns specified customer 1`] = `
Object {
  "email": "dimitris.klouvas@gmail.com",
  "first_name": "Dimitris",
  "id": Anything,
  "last_name": "Klouvas",
}
`;
